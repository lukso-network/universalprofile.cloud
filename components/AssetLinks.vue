<script setup lang="ts">
import type { ButtonSize } from '@lukso/web-components'

type Props = {
  asset?: Asset
  buttonSize?: ButtonSize
  withoutTitle?: boolean
}

const props = withDefaults(defineProps<Props>(), {
  asset: undefined,
  buttonSize: 'medium',
  withoutTitle: false,
})
const { isMobile } = useDevice()
const { formatMessage } = useIntl()
const isLoaded = computed(() => props.asset && !props.asset?.isMetadataLoading)
const links = computed(() => props.asset?.resolvedMetadata?.links)
</script>

<template>
  <template v-if="isLoaded">
    <div v-if="links?.length">
      <div v-if="!withoutTitle" class="heading-inter-14-bold pb-2">
        {{ formatMessage('asset_links_title') }}
      </div>
      <div class="grid grid-cols-1 gap-2 sm:flex sm:flex-wrap">
        <div v-for="(link, index) in links" :key="index" class="inline-flex">
          <lukso-button
            variant="secondary"
            :size="buttonSize"
            is-link
            :href="link.url"
            :is-full-width="isMobile ? true : undefined"
            custom-class="paragraph-inter-14-medium"
          >
            <div class="flex w-full items-center text-left">
              {{ link.title || link.url }}
              <lukso-icon name="link-1" size="small" class="ml-2"></lukso-icon>
            </div>
          </lukso-button>
        </div>
      </div>
    </div>
  </template>
  <AppPlaceholderSection v-else slot-class="flex gap-4">
    <AppPlaceholderLine
      :class="{
        'h-[48px] w-[170px]': buttonSize === 'medium',
        'h-[24px] w-[90px]': buttonSize === 'small',
      }"
    />
    <AppPlaceholderLine
      :class="{
        'h-[48px] w-[170px]': buttonSize === 'medium',
        'h-[24px] w-[90px]': buttonSize === 'small',
      }"
    />
  </AppPlaceholderSection>
</template>
