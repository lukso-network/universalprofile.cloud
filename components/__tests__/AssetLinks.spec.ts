import { renderSuspended } from '@nuxt/test-utils/runtime'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import AssetLinks from '@/components/AssetLinks.vue'

describe('AssetLinks', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('should render loading state for medium buttons', async () => {
    const component = await renderSuspended(AssetLinks)

    expect(component.html()).toMatchSnapshot()
  })

  it('should render loading state for small buttons', async () => {
    const component = await renderSuspended(AssetLinks, {
      props: {
        buttonSize: 'small',
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render links for medium buttons', async () => {
    const component = await renderSuspended(AssetLinks, {
      props: {
        buttonSize: 'small',
        asset: {
          resolvedMetadata: {
            links: [
              {
                url: 'https://link1.example.com',
                title: 'Link 1',
              },
              {
                url: 'https://link2.example.com',
                title: 'Link 2',
              },
            ],
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render links for small buttons', async () => {
    const component = await renderSuspended(AssetLinks, {
      props: {
        buttonSize: 'small',
        asset: {
          resolvedMetadata: {
            links: [
              {
                url: 'https://link1.example.com',
                title: 'Link 1',
              },
              {
                url: 'https://link2.example.com',
                title: 'Link 2',
              },
            ],
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render for no links', async () => {
    const component = await renderSuspended(AssetLinks, {
      props: {
        asset: {
          resolvedMetadata: {
            links: [],
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render without title', async () => {
    const component = await renderSuspended(AssetLinks, {
      props: {
        withoutTitle: true,
        asset: {
          resolvedMetadata: {
            links: [
              {
                url: 'https://link1.example.com',
                title: 'Link 1',
              },
            ],
          },
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
