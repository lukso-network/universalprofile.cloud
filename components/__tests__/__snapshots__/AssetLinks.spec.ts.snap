// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetLinks > should render for no links 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;

exports[`AssetLinks > should render links for medium buttons 1`] = `
"<div id="test-wrapper">
  <div>
    <div class="heading-inter-14-bold pb-2">Links</div>
    <div class="grid grid-cols-1 gap-2 sm:flex sm:flex-wrap">
      <div class="inline-flex">
        <lukso-button variant="secondary" size="small" is-link="" href="https://link1.example.com" custom-class="paragraph-inter-14-medium">
          <div class="flex w-full items-center text-left">Link 1 <lukso-icon name="link-1" size="small" class="ml-2"></lukso-icon>
          </div>
        </lukso-button>
      </div>
      <div class="inline-flex">
        <lukso-button variant="secondary" size="small" is-link="" href="https://link2.example.com" custom-class="paragraph-inter-14-medium">
          <div class="flex w-full items-center text-left">Link 2 <lukso-icon name="link-1" size="small" class="ml-2"></lukso-icon>
          </div>
        </lukso-button>
      </div>
    </div>
  </div>
</div>"
`;

exports[`AssetLinks > should render links for small buttons 1`] = `
"<div id="test-wrapper">
  <div>
    <div class="heading-inter-14-bold pb-2">Links</div>
    <div class="grid grid-cols-1 gap-2 sm:flex sm:flex-wrap">
      <div class="inline-flex">
        <lukso-button variant="secondary" size="small" is-link="" href="https://link1.example.com" custom-class="paragraph-inter-14-medium">
          <div class="flex w-full items-center text-left">Link 1 <lukso-icon name="link-1" size="small" class="ml-2"></lukso-icon>
          </div>
        </lukso-button>
      </div>
      <div class="inline-flex">
        <lukso-button variant="secondary" size="small" is-link="" href="https://link2.example.com" custom-class="paragraph-inter-14-medium">
          <div class="flex w-full items-center text-left">Link 2 <lukso-icon name="link-1" size="small" class="ml-2"></lukso-icon>
          </div>
        </lukso-button>
      </div>
    </div>
  </div>
</div>"
`;

exports[`AssetLinks > should render loading state for medium buttons 1`] = `
"<div id="test-wrapper">
  <div>
    <div class="mb-3 h-[17px] w-1/2 animate-pulse rounded-4 bg-neutral-90"></div>
    <div class="flex gap-4">
      <div class="animate-pulse rounded-4 bg-neutral-90 h-[48px] w-[170px]"></div>
      <div class="animate-pulse rounded-4 bg-neutral-90 h-[48px] w-[170px]"></div>
    </div>
  </div>
</div>"
`;

exports[`AssetLinks > should render loading state for small buttons 1`] = `
"<div id="test-wrapper">
  <div>
    <div class="mb-3 h-[17px] w-1/2 animate-pulse rounded-4 bg-neutral-90"></div>
    <div class="flex gap-4">
      <div class="animate-pulse rounded-4 bg-neutral-90 h-[24px] w-[90px]"></div>
      <div class="animate-pulse rounded-4 bg-neutral-90 h-[24px] w-[90px]"></div>
    </div>
  </div>
</div>"
`;

exports[`AssetLinks > should render without title 1`] = `
"<div id="test-wrapper">
  <div>
    <!--v-if-->
    <div class="grid grid-cols-1 gap-2 sm:flex sm:flex-wrap">
      <div class="inline-flex">
        <lukso-button variant="secondary" size="medium" is-link="" href="https://link1.example.com" custom-class="paragraph-inter-14-medium">
          <div class="flex w-full items-center text-left">Link 1 <lukso-icon name="link-1" size="small" class="ml-2"></lukso-icon>
          </div>
        </lukso-button>
      </div>
    </div>
  </div>
</div>"
`;
